iquip/apps/monitor.py

```diff
@@ -209,6 +211,140 @@ def run(self):
+        self.slider.setTickInterval(self._unit)
+        self.slider.setTickPosition(QSlider.TicksAbove)
+        minVoltageLabel = QLabel(f"Min: {minVoltage}V", self)
+        minVoltageLabel.setAlignment(Qt.AlignLeft | Qt.AlignVCenter)
+        maxVoltageLabel = QLabel(f"Max: {maxVoltage}V", self)
+        maxVoltageLabel.setAlignment(Qt.AlignRight | Qt.AlignVCenter)
+        self.voltageSpinBox = QDoubleSpinBox(self)
+        self.voltageSpinBox.setAlignment(Qt.AlignVCenter)
+        self.voltageSpinBox.setSuffix("V")
+        self.voltageSpinBox.setMinimum(minVoltage)
+        self.voltageSpinBox.setMaximum(maxVoltage)
+        self.voltageSpinBox.setDecimals(ndecimals)
+        self.setButton = QPushButton("Set")
+        self._sliderChanged(self.slider.value())
+        # layout
+        infoLayout = QHBoxLayout()
+        infoLayout.addWidget(nameLabel)
+        infoLayout.addWidget(deviceLabel)
+        infoLayout.addWidget(channelLabel)
+        sliderInfoLayout = QHBoxLayout()
+        sliderInfoLayout.addWidget(minVoltageLabel)
+        sliderInfoLayout.addWidget(self.voltageSpinBox)
+        sliderInfoLayout.addWidget(maxVoltageLabel)
+        layout = QVBoxLayout(self)
+        layout.addLayout(infoLayout)
+        layout.addWidget(self.slider)
+        layout.addLayout(sliderInfoLayout)
+        layout.addWidget(self.setButton)
+        # signal connection
+        self.slider.valueChanged.connect(self._sliderChanged)
+        self.voltageSpinBox.valueChanged.connect(self._spinBoxChanged)
+        self.setButton.clicked.connect(self._setButtonClicked)
+
+    @pyqtSlot(int)
+    def _sliderChanged(self, value: int):
+        """The slider value is changed.
+        
+        Args:
+            value: Current slider value.
+        """
+        self.voltageSpinBox.setValue(value / self._unit)
+
+    @pyqtSlot(float)
+    def _spinBoxChanged(self, value: float):
+        """The voltageSpinBox value is changed.
+        
+        Args:
+            value: Current voltageSpinBox value.
+        """
+        self.slider.setValue(int(value * self._unit))
+
+    @pyqtSlot()
+    def _setButtonClicked(self):
+        """The setButton is clicked."""
+        self.voltageSet.emit(self.slider.value() / self._unit)
+
+
+class DACControllerFrame(QWidget):
+    """Frame for monitoring and controlling DAC channels.
+    
+    Attributes:
+        dacWidgets: Dictionary with DAC controller widgets.
+          Each key is a DAC channel name, and its value is the corresponding DACControllerWidget.
+    """
+
+    def __init__(
+        self,
+        dacInfo: Dict[str, Dict[str, Union[float, str]]],
+        numColumns: int = 4,
+        parent: Optional[QWidget] = None
+    ):
+        """Extended.
+        
+        Args:
+            dacInfo: Dictionary with DAC channels info.
+              Each key is a DAC channel name, and its value is a dictionary with DAC info.
+              About this dictionary, see DACControllerWidget.__init__().
```

Comment:
I think saying that it is given as keyword arguments to `DACControllerWidget.__init__()` would be clearer.